docs/manual/src/udl/ext_types_wrapped.md

```diff
@@ -57,13 +83,83 @@ Because `UniffiCustomTypeWrapper` is defined in each crate, this means you can w
 if they are not defined in your crate - see the 'wrapper_types' example which demonstrates
 how `serde_json::Value` can be wrapped.
 
-## Custom Foreign Wrappers
+## Bindings example
+
+Suppose you have Rust struct with recently renamed fields:
+
+```rust
+pub struct User {
+  full_name: String,
+  age: Int32,
+}
+```
+
+But your Kotlin code is still using the old field names:
+
+```kotlin
+data class LegacyUser(
+  var name: String,
+  var years_old: Int,
+}
+```
+
+We again start by declaring this type in our UDL via a `typedef` with a `Wrapped` attribute.
+
+```idl
+[Wrapped]
+typedef User LegacyUser;
+```
+
+This time however, we will update the Kotlin section of the `uniffi.toml` file:
+
+```toml
+[bindings.kotlin.wrappers.LegacyUser]
+# Name of the Kotlin class that will wrap `User`
+type_name = "LegacyUser"
+# Function that converts `User` to `LegacyUser`
+wrap = "wrapUser"
+# Function that converts `LegacyUser` to `User`
+unwrap = "unwrapUser"
+# Where we can import those functions from
+imports = [
+  "my.wrappers.package.wrapUser",
+  "my.wrappers.package.unwrapUser",
+]
+```
+
+`my.wrappers.package` would have code like this:
+
+```kotlin
+fun wrapUser(user: User) = LegacyUser(
+   name = user.fullName,
+   yearsOld = user.age,
+)
+
+fun unwrapUser(user: LegacyUser) = User(
+   fullName = user.name,
+   age = user.yearsOld,
+)
+```
+
+## Wrapping on both the scaffolding and bindings side
+
+Of course, it's possible to wrap on both the scaffolding and bindings side.
+This allows you to define types that feel like new builtin types.  The
+canonical example here is a `JsonObject` type that leverages `string` to
+marshall data across the FFI.  Check out the Python example in
+`examples/wrapper-types` for how this could work.
+
+## uniffi.toml configuration
 
-In the example above, the foreign bindings just see the "wrapped" value - eg, the bindings will
-get an integer for the `Handle`.
+Wrapper types require some configuration in `uniffi.toml`.  Here's how it works:
 
-We intend on making it possible to customize this - for example, `serde_json::Value` could be
-transformed into a suitable JSON object in the bindings.
+* Create a `[scaffolding.wrappers.{WrapperTypeName}]` table to enable wrapping on the scaffolding side.  This only has one key:
+  * `type_name`: Rust type name for the wrapper.
+* Create a `[bindings.{language}.wrappers.{WrapperTypeName}]` table to enable wrapping on a bindings side.  This has several subkeys:
+  * `type_name` (Typed languages only): Type/class name for the wrapped type
+  * `wrap`: Function to wrap the UDL type, converting it to the wrapped type.
+  * `unwrap`: Function to unwrap the wrapped type, converting it to the UDL type.
```

Comment:
I don't disagree, but I haven't been able to get anything better.  One idea I was thinking about was `from_inner`, `into_inner` instead of `wrap` / `unwrap`.

For `Type::Wrapped`, I wonder if we could eventually making that just `Type::Typedef` and allow any type to be wrapped.  I don't know if this fixes the naming issue though, it's just that a different thing would be labeled "wrapped".

Reply:
I like the `from` and `into` formulation; I'd been think `from_raw` or `from_lower`.

I like the `Type::Typedef`; though it may be a  bit too general — we may find some clever use for `typedef`s in the future.